pontus/week1/ass2/index.html

```diff
@@ -1,30 +1,160 @@
+		<script src="http://fb.me/react-dom-0.14.7.min.js"></script>
+		<!-- <script src="http://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.4.4/babel.min.js"></script> -->
+		<script src="http://fb.me/JSXTransformer-0.12.2.js"></script>
+	</head>
+	<body>
+
+		<div id="app"></div>
+
+		<!-- 1. Create an inputfield that doesn't accept any numbers -->
+		<!-- 2. Find a way to submit and print the message -->
+		<!-- 3. Validation errors in inputfields should produce an error-message rather than a hello morld message -->
+		<!-- 4. Any error-message should be printed in red color -->
+		<!-- 5. Provide a function that returns all messages ever printed -->
+		<!-- 6. This must be solved by using only native Javascript and react -->
+		<!-- 7. All markup must be produced with react -->
+		<!-- 8. All messages must be printed in the order they where submitted -->
+
+
+
+		<script type="text/jsx" >
+var Wrapper = React.createClass({ 
+			getInitialState: function() { 
+
+			return { 
+					text: "",
+					messages : [],
+					error: false
+				} 
+
+			},
+
+			handleCorrectInput: function(val) { 
+
+				this.state.messages.push(val);
+
+				// update the state.
+				this.setState({ 
+					error: false,
+					text: val
+				});
+			},
+
+			handleIncorrectInput: function(error) { 
+
+				this.setState({ 
+					text:error, 
+					error: true
+				});
+
+
+			},
+
+			showAll: function() { 
+
+			var str = this.state.messages.join('\n');
+
+				this.setState({ 
+					text : str
+				});
+
+			},
+
+			render: function() { 
+
+
+				return ( 
+						<div>
+							<InputField handleIncorrectInput={this.handleIncorrectInput} handleCorrectInput={this.handleCorrectInput} text={this.state.text} />
+							<OutputField error={this.state.error} showAll={this.showAll} text={this.state.text} messages={this.state.messages} />
+						</div>
+				);
+			}
+		});
+
+
+
+
+		// input field
+
+		var InputField = React.createClass({ 
+
+			handleClick() { 
+
+				var re = /\d/;
+				var val = this.refs.inputField.value;
+
+				if (re.test(val)) { 
+					var error = "error: must use only alphabetic characters.";
+
+
+					// callback to method in wrapper
+					this.props.handleIncorrectInput(error);
+
+				} else { 
+
+					// callback to method in wrapper
+					this.props.handleCorrectInput(val);
+				}
+
+			},
+
+			render: function() { 
+				return ( 
+					<div>
+						<input type="text" ref='inputField'/>
+						<button onClick={this.handleClick}>submit</button>
+					</div>
+				);
+			}
+		});
+
+		var OutputField = React.createClass({ 
+
+			showAllClick: function() { 
+
+
+			// callback to wrapper.
+			this.props.showAll();
+
+			},
+
+			render: function() { 
+
+				var outputStyle = (this.props.error) ? {color: 'red'} : {color: 'black' }
+
+				return ( 
+					<div>
+						<textarea type="text" style={outputStyle}  value={this.props.text}/>
+						<button onClick={this.showAllClick}>show all</button>
```

Comment:
You do not need another function. You could have simply passed in the props function `onClick={this.props.showAll}`